specs/cli/plugin-add.md

```diff
@@ -0,0 +1,185 @@
+# kiota plugin add
+
+## Description 
+
+`kiota plugin add` allows a developer to add a new plugin to the `workspace.json` file. If no `workspace.json` file is found, a new `workspace.json` file would be created in the `.kiota` directory under the current working directory. The command will add a new entry to the `plugins` section of the `workspace.json` file. Once this is done, a local copy of the OpenAPI description is generated and kept in the `.kiota/plugins` folder. If a plugin or client with the same name already exists, the command will fail and display an actionable error message.
+
+When executing, a new plugin entry will be added and will use the `--plugin-name` parameter as the key for the map. When loading the OpenAPI description, it will store the location of the description in the `descriptionLocation` property. If `--include-path` or `--exclude-path` are provided, they will be stored in the `includePatterns` and `excludePatterns` properties respectively.
+
+Every time a plugin is added, a copy of the OpenAPI description file will be stored in the `./.kiota/plugins` folder. The OpenAPI will be named using the plugin name `{plugin-name}.json|yaml`. This will allow the CLI to detect changes in the description and avoid downloading the description again if it hasn't changed.
```

Comment:
Also we're now creating a directory per client/consumer in anticipation for overlays and other multi-document based generation support. `.kiota/clients/clientName/description.yml|json`

Reply:
I'd still keep a sub-segment like "descriptions" because with multiple clients which points to descriptions with multiple files, it'll become messy quickly

Reply:
`apis` ?

Reply:
sure, whatever name you'd like, just don't make me change it afterwards 😉
documents would also be a good candidate since this directory would contain documents (OpenAPI, overlays, etc..)

Reply:
I can deal with documents!

Reply:
so documents it is!
@maisarissi can you update it across the spec when you have a couple of minutes please?

Reply:
Just did. Please check the updated spec and resolve the conversation if I got it right :)